docs/guides/interactivity.md

```diff
@@ -1,63 +1,104 @@
+
+> Note: Mesop has a built-in loading indicator at the top of the page for all events.
 
 ```python
-@me.stateclass
-class State:
-  clicks: int
+--8<-- "mesop/examples/docs/loading.py"
 ```
 
-Each user session, which is equivalent to a browser session, will have its own state instance. Everything in a state class, recursively, must be serializable. This is because, under the hood, Mesop is sending the state back and forth between the server and browser client.
+In this example, our event handler is a Python generator function. Each `yield` statement yields control back to the Mesop framework and executes a render loop which results in a UI update.
 
-### Nested State
+Before the first yield statement, we set `is_loading` to True on state so we can show a spinner while the user is waiting for the slow API call to complete.
 
-You can also have classes inside of a state class as long as everything is serializable:
+Before the second (and final) yield statement, we set `is_loading` to False, so we can hide the spinner and then we add the result of the API call to state so we can display that to the user.
+
+> Tip: you must have a yield statement as the last line of a generator event handler function. Otherwise, any code after the final yield will not be executed.
+
+## Streaming
+
+This example builds off the previous Loading example and makes our event handler a generator function so we can incrementally update the UI.
 
 ```python
-class NestedState:
-  val: int
+--8<-- "mesop/examples/docs/streaming.py"
+```
 
+## Troubleshooting
+
+### User input race condition
+
+If you notice a race condition with user input (e.g. [input](../components/input.md) or [textarea](../components/textarea.md)) where sometimes the last few characters typed by the user is lost, you are probably unnecessarily setting the value of the component.
+
+See the following example using this **anti-pattern** :warning::
+
+```py title="Bad example"
 @me.stateclass
 class State:
-  nested: NestedState
+  input_value: str
 
 def app():
   state = me.state(State)
+  me.input(value=state.input_value, on_input=on_input)
+
+def on_input(event: me.InputEvent):
+  state = me.state(State)
+  state.input_value = event.value
 ```
 
-> Note: you only need to decorate the top-level state class with `@me.stateclass`. All the nested state classes will automatically be wrapped.
+The problem is that the input value now has a race condition because it's being set by two sources:
 
-## Slow / async patterns
+1. The server is setting the input value based on state.
+2. The client is setting the input value based on what the user is typing.
 
-These are patterns for dealing with common use cases such as calling a slow blocking API call or a streaming API call.
+The way to fix this is by *not* setting the input value from the server.
 
-### Loading
+The above example **corrected** would look like this :white_check_mark::
```

Comment:
[nit] May also want to include an example of setting a initial value for an input or updating the initial value. Also clearing using input may be good things to have.

Reply:
let's add this later (this was a copy & paste from before)